collectors/freebsd.plugin/freebsd_sysctl.c

```diff
@@ -1044,9 +1044,9 @@ int do_system_ram(int update_every, usec_t dt) {
         rrddim_set_by_pointer(st, rd_free,     vmmeter_data.v_free_count);
         rrddim_set_by_pointer(st, rd_active,   vmmeter_data.v_active_count);
         rrddim_set_by_pointer(st, rd_inactive, vmmeter_data.v_inactive_count);
-        rrddim_set_by_pointer(st, rd_wired,    vmmeter_data.v_wire_count * system_pagesize - zfs_arcstats_shrinkable_cache_size_bytes);
+        rrddim_set_by_pointer(st, rd_wired,    (unsigned long long)vmmeter_data.v_wire_count * (unsigned long long)system_pagesize - zfs_arcstats_shrinkable_cache_size_bytes);
 #if __FreeBSD_version < 1200016
-        rrddim_set_by_pointer(st, rd_cache,    vmmeter_data.v_cache_count * system_pagesize + zfs_arcstats_shrinkable_cache_size_bytes);
+        rrddim_set_by_pointer(st, rd_cache,    (unsigned long long)vmmeter_data.v_cache_count * (unsigned long long)system_pagesize + zfs_arcstats_shrinkable_cache_size_bytes);
 #else
```

Comment:
Shouldn't the cast be to `collected_number` type?

Reply:
I think there is the implicit casting: calculation result => collected_number when passed to the func. Not a specialist (not c dev).

Reply:
Yep, that's what I had in mind. You may get a compiler warning in some cases.

Reply:
> some compilers can create warnings, mainly the oldest one

is not a strong reason, I am not "fixing" it in this PR. If you check other "rrddim_set_by_pointer" calls in this collector you see there are no casting to "collected_number". If you believe this is needed consider creating a separate PR.